sdcm/nemesis.py

```diff
@@ -1457,8 +1457,78 @@ def target_node_reported_un_by_others():
         self.tester.verify_no_drops_and_errors(starting_from=start_time)
 
     def disrupt_snapshot_operations(self):
+        """
+        Extend this nemesis to run 'nodetool snapshot' more options including multiple tables.
+        Random choose between:
+        - create snapshot of all keyspaces
+        - create snapshot of one keyspace
+        - create snapshot of few keyspaces
+        - create snapshot of few tables, including MVs
+        """
+        def _full_snapshot():
+            self.log.info("Take all keyspaces snapshot")
+            return 'snapshot'
+
+        def _ks_snapshot(one_ks):
+            self.log.info(f"Take {'one keyspace' if one_ks else 'few keyspaces'} snapshot")
+            # Prefer to take snapshot of test keyspace. Try to find it
+            ks_cf = self.cluster.get_non_system_ks_cf_list(db_node=self.target_node)
+            if not ks_cf:
+                # If test table wasn't found - take system keyspace snapshot
+                ks_cf = self.cluster.get_any_ks_cf_list(db_node=self.target_node)
+
+            if one_ks:
+                keyspace_table = random.choice(ks_cf)
+                keyspace, _ = keyspace_table.split('.')
+            else:
+                keyspaces = list(set([ks.split('.')[0] for ks in ks_cf]))
+                if len(keyspaces) == 1:
+                    ks_cf = self.cluster.get_any_ks_cf_list(db_node=self.target_node)
+                    keyspaces = list(set([ks.split('.')[0] for ks in ks_cf]))
+                keyspace = ','.join(keyspaces)
+
+            return f'snapshot -kc {keyspace}'
+
+        def _few_tables():
+            def get_ks_with_few_tables():
+                ks_occurrences = Counter([ks.split('.')[0] for ks in ks_cf])
+                repeated_ks = [ks for ks, count in ks_occurrences.items() if count > 1]
+                return repeated_ks
+
+            self.log.info("Take few tables snapshot")
+            # Prefer to take snapshot of test table. Try to find it
+            ks_cf = self.cluster.get_non_system_ks_cf_list(db_node=self.target_node)
+
+            if not ks_cf or len(ks_cf) == 1:
+                # If test table wasn't found - take system table snapshot
+                ks_cf = self.cluster.get_any_ks_cf_list(db_node=self.target_node)
+
+            ks_with_few_tables = get_ks_with_few_tables()
+            if not ks_with_few_tables:
+                ks_cf = self.cluster.get_any_ks_cf_list(db_node=self.target_node)
+                ks_with_few_tables = get_ks_with_few_tables()
```

Comment:
I don't understand how `get_ks_with_few_tables` will find ks if it didn't find in the line before the "if" condition (L:1506)

Reply:
First time I try to find keyspace with few non-system tables using get_NON_SYSTEM_ks_cf_list (L:1500):
```
# Prefer to take snapshot of test table. Try to find it
ks_cf = self.cluster.get_non_system_ks_cf_list(db_node=self.target_node)
```

If keyspace with few tables is not found - I take system keyspace using get_ANY_ks_cf_list (L:1506):
```
ks_cf = self.cluster.get_any_ks_cf_list(db_node=self.target_node)
```